### PR TITLE
refactor(variable-selector): bind variable data in use-change situation

### PR DESCRIPTION
**Pull request by piggggggggy**
### To Reviewers
- [ ] Skip (`style`, `chore` ONLY)
- [x] Not that difficult

### Type of Change
- [ ] New feature
- [ ] Bug fixes
- [ ] Feature improvement
- [x] Refactor
- [ ] Others (performance improvement, refactoring, CI/CD, etc.)

### Checklist
- [x] `Error / Warning / Lint / Type`

### Description
SSIA

- If toggle variable use to false, that variable property's selected options will reset in variable options data.

https://user-images.githubusercontent.com/83635051/210554322-4c0e9199-c4eb-469f-b41c-48d5d5730652.mov

### CODE REVIEW

File: src/services/dashboards/dashboard-customize/DashboardCustomizePage.vue [hunk 1]
```diff
@@ -219,6 +219,10 @@ const handleSave = async () => {
 const handleChangeVariable = (variables: DashboardVariablesSchema['properties'], order?: string[]) => {
     variableState.variableProperties = variables;
     if (order) variableState.order = order;
+    variableState.order.forEach((d) => {
+        if (variables[d].use) return;
+        if (variableState.variableData[d]) delete variableState.variableData[d];
+    });
```
Comment (minseolee) [inline src/services/dashboards/dashboard-customize/DashboardCustomizePage.vue:225]: Is it ok to use original object?
Below code is just a small suggestion!
```suggestion
    variableState.order.forEach((d) => {
        if (variables[d].use) return;
        if (variableState.variableData[d]) {
            const _variableData = { ...variableState.variableData };
            delete _variableData[d];
            variableState.variableData = _variableData;
        }
    });
```

**Review by minseolee — COMMENTED** [review]
Left a comment! 😉

File: src/services/dashboards/dashboard-customize/DashboardCustomizePage.vue (hunk 1 above)
Comment (yuda110) [inline src/services/dashboards/dashboard-customize/DashboardCustomizePage.vue:225]: I agree with @minseolee 's opinion! I prefer to create the copied object first.

```suggestion
   const _variableData = cloneDeep(variableState.variableData);
    variableState.order.forEach((d) => {
        if (variables[d].use) return;
        if (variableState.variableData[d]) delete _variableData[d];
    });
    variableState.variableData = _variableData;
```

**Review by yuda110 — APPROVED** [review]
LGTM! I left a comment

File: src/services/dashboards/dashboard-customize/DashboardCustomizePage.vue (hunk 1 above)
Comment (yuda110) [reply src/services/dashboards/dashboard-customize/DashboardCustomizePage.vue:225]: I agree with @minseolee 's opinion! I prefer to create the copied object first.

```suggestion
   const _variableData = cloneDeep(variableState.variableData);
    variableState.order.forEach((d) => {
        if (variables[d].use) return;
        if (variableState.variableData[d]) delete _variableData[d];
    });
    variableState.variableData = _variableData;
```

**Pull request merged**